pulumi/backuprunner/src/scaling.ts

```diff
@@ -0,0 +1,63 @@
+import k8s from '@kubernetes/client-node';
+
+export const delay = (time: number) => {
+  return new Promise((resolve) => setTimeout(resolve, time));
+};
+
+export const scaleStatefulSet = async (
+  k8sAppsClient: k8s.AppsV1Api,
+  name: string,
+  namespace: string,
+  count: number,
+  waitToFinish: boolean
+) => {
+  console.log(`Scaling StatefulSet ${namespace}.${name} to ${count} replicas`);
+
+  const { body } = await k8sAppsClient.readNamespacedStatefulSet(name, namespace);
+
+  if(typeof(body.spec?.replicas) === "string"){
+    body.spec.replicas = count
+  }else{
+    console.warn(`Invalid replica count: ${body.spec?.replicas}`)
+  }
```

Comment:
I would go this route for narrowing:
```suggestion
  if (!body.spec) throw new Error(`No spec found for StatefulSet: ${namespace}.${name}`)

  body.spec.replicas = count
```
If we don't have a spec to modify, consider the scale/backup invalid. This lets ts know `spec` is no undefined, then we can just set `count` as desired.

`replicas` is `number | undefined` so `typeof string` should be invalid regardless

Reply:
Missed the wrong type check, thx